util/src/main/java/io/kubernetes/client/informer/SharedInformer.java

```diff
@@ -54,4 +55,19 @@
   // store. The value returned is not synchronized with access to the underlying store and is not
   // thread-safe.
   String lastSyncResourceVersion();
+
+  /**
+   * The TransformFunc is called for each object which is about to be stored. This function is
+   * intended for you to take the opportunity to remove, transform, or normalize fields. One use
+   * case is to strip unused metadata fields out of objects to save on RAM cost.
+   *
+   * <p>Must be set before starting the informer.
+   *
+   * <p>Note: Since the object given to the handler may be already shared with other goroutines, it
+   * is advisable to copy the object being transform before mutating it at all and returning the
+   * copy to prevent data races.
+   *
+   * @param transformFunc the transform function
+   */
+  void setTransformFunc(Function<KubernetesObject, KubernetesObject> transformFunc);
```

Comment:
lets define a functional interface named `TransformFunc` to align w/ the upsteam, also the original go function returns an error upon failure, so i think we should declare a checked exception e.g. `throws ObjectTransformException`. the unit test should also cover the failure path when the checked exception is thrown.

Reply:
> also the original go function returns an error upon failure, so i think we should declare a checked exception e.g. `throws ObjectTransformException`.

The exception handling references `addIndexers`, which returns an error in client-go while in client-java it doesn't declare an exception
https://github.com/kubernetes/kubernetes/blob/v1.23.4/staging/src/k8s.io/client-go/tools/cache/shared_informer.go#L189
https://github.com/kubernetes-client/java/blob/a5a7ce94abab72391b6425436e7e5af01a4c5854/util/src/main/java/io/kubernetes/client/informer/impl/DefaultSharedIndexInformer.java#L287-L292
Do you mean something like this?
```java
public interface TransformFunc {

  /**
   * @param the original object to be transformed
   * @return the transformed object
   */
  KubernetesObject transform(KubernetesObject object) throws ObjectTransformException;
}
```

Reply:
> Do you mean something like this?

yes, thats exactly what i thought, as for `addIndexers`, i think we can keep it, it's the legacy code and it's working.

Reply:
OK, fixed